mrmustard/lab_dev/circuit_components.py

```diff
@@ -247,6 +287,12 @@ def __matmul__(self, other: CircuitComponent) -> CircuitComponent:
 
         return CircuitComponent._from_attributes("", representation_ret, wires_ret)
 
+    def __lshift__(self, other: CircuitComponent):
+        r"""
+        Projects this component onto another component by using self's ``>>`` on ``other.dual``.
+        """
+        return self >> other.dual
```

Comment:
```suggestion
        return (other.dual >> self.dual).dual
```

Reply:
I actually think that the correct one is: `return (other >> self.dual).dual`